registrations/tests/test_commands.py

```diff
@@ -128,6 +128,48 @@ def test_anonymize_past_signups():
     assert past_signup.anonymization_time is not None
 
 
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "enrolment_time_delta,event_time_delta,should_anonymize",
+    [
+        (0, -31, False),
+        (31, -31, False),
+        (-29, -31, False),
+        (-31, -31, True),
+        (-31, 0, False),
+        (-31, 31, False),
+        (-31, -29, False),
+    ],
+)
+def test_anonymize_if_enrolment_end_time_and_end_time_are_over_30_days_in_past(
+    enrolment_time_delta, event_time_delta, should_anonymize
+):
+    event = EventFactory(end_time=localtime() + timedelta(days=event_time_delta))
+    registration = RegistrationFactory(
+        event=event,
+        enrolment_end_time=localtime() + timedelta(days=enrolment_time_delta),
+    )
+    signup_group = SignUpGroupFactory(registration=registration)
+    signup_in_group = SignUpFactory(
+        registration=registration, signup_group=signup_group
+    )
+    signup = SignUpFactory(registration=registration)
+
+    call_command("anonymize_past_signups")
+
+    signup_group.refresh_from_db()
+    signup_in_group.refresh_from_db()
+    signup.refresh_from_db()
+    if should_anonymize:
```

Comment:
Highly suggest on splitting tests in two separate test. Other option would be using expected value in the parameters instead of bool  which would lead removing the `if else` conditions and using the expect value in the assert.